lib/sisimai/bite/email/sendgrid.rb

```diff
@@ -0,0 +1,223 @@
+module Sisimai::Bite::Email
+  module SendGrid
+    # Sisimai::Bite::Email::SendGrid parses a bounce email which created by
+    # SendGrid. Methods in the module are called from only Sisimai::Message.
+    class << self
+    # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/SendGrid.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :'from'        => %r/\AMAILER-DAEMON\z/,
+        :'return-path' => %r/\A[<]apps[@]sendgrid[.]net[>]\z/,
+        :'subject'     => %r/\AUndelivered Mail Returned to Sender\z/,
+      }
+      Re1 = {
+        :begin  => %r/\AThis is an automatically generated message from SendGrid[.]\z/,
+        :error  => %r/\AIf you require assistance with this, please contact SendGrid support[.]\z/,
+        :rfc822 => %r|\AContent-Type: message/rfc822|,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'SendGrid: http://sendgrid.com/'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+
+      # Return-Path: <apps@sendgrid.net>
+      # X-Mailer: MIME-tools 5.502 (Entity 5.502)
+      def headerlist;  return ['Return-Path', 'X-Mailer']; end
```

Comment:
__[rubocop]__
* Style/SingleLineMethods: Avoid single-line method definitions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-single-line-methods)
* Style/Semicolon: Do not use semicolons to terminate expressions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-semicolon)
* Layout/TrailingWhitespace: Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

<!-- Comment by [SideCI](https://sideci.com) -->